### PR TITLE
Use Github Actions instead of TravisCI

### PR DESCRIPTION
**Pull request by dolfinus**
Just like absinthe and absinthe_relay let's move from TravisCI to Github Actions.
Here is an example of a job run: https://github.com/dolfinus/dataloader/runs/2170029231?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        env:
+          POSTGRES_DB: dataloader_test
+          POSTGRES_PASSWORD: postgres
+
+    strategy:
+      matrix:
+        elixir:
+          - '1.7'
+          - '1.8'
+          - '1.9'
```
Comment (benwilson512) [inline .github/workflows/test.yml:28]: Absinthe tends to only support the last 2 Elixir versions, so we can remove 1.7-1.9. This does need to be updated in our `mix.exs` probably.

Comment (dolfinus) [reply .github/workflows/test.yml]: Done.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    - name: Restore deps cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          deps
+          _build
+        key: ${{ runner.os }}-deps-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}-git-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-deps-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
```
Comment (binaryseed) [inline .github/workflows/test.yml:50]:
```suggestion
          deps-${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
```
small tweak if you don't mind

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    - name: Run unit tests
+      run: |
+        mix ecto.setup
+        mix test
+      env:
+        MIX_ENV: test
+
+    - name: Run dialyzer
+      run: mix dialyzer
```
Comment (binaryseed) [inline .github/workflows/test.yml:84]: missing linebreak

File: .gitignore [hunk 4]
```diff
@@ -24,3 +24,7 @@ dataloader-*.tar
 
 # Temporary files for e.g. tests.
 /tmp/
+
+# Ignore Dialyzer .plts files
+/priv/plts/*.plt
+/priv/plts/*.plt.hash
```
Comment (binaryseed) [inline .gitignore:30]: missing linebreak

File: .github/workflows/test.yml (hunk 2 above)
Comment (dolfinus) [reply .github/workflows/test.yml]: Fixed.

**Review by binaryseed — CHANGES_REQUESTED** [review]
I'm seeing `priv/plts/dialyxir_erlang-23.2.5_elixir-1.11.3.plt` and another `plt` file snuck in this branch...

**Comment by dolfinus** [comment]
Oh, my bad, I've fixed .gitignore but didn't saw that these files are already in the commit. It is removed now.

**Review by binaryseed — APPROVED** [review]

**Comment by binaryseed** [comment]
Thanks!

**Pull request merged**